packages/flutter_tools/lib/src/commands/update_packages.dart

```diff
@@ -1379,13 +1391,14 @@ File _pubspecFor(Directory directory) {
 
 /// Generates the source of a fake pubspec.yaml file given a list of
 /// dependencies.
-String _generateFakePubspec(
+@visibleForTesting
+String generateFakePubspec(
```

Comment:
I have a general question about private methods. Is it ok to make them public as long as you add the visibleForTesting? Quoting Dan in this [answer](https://stackoverflow.com/questions/21657315/how-to-test-a-private-function-in-dart) "if it's something worth testing, it's something that's worth being public". Regardless of the answer this is a nit/curiosity

Reply:
I think it's a nice goal to strive for to only test public methods. However, this command would be incredibly difficult to end to end test. Also, this synthetic package gets created and deleted during the running of the command, so it would not really be practical to test after the command completed (the behavior we're interested in is also dependent on a heuristic algorithm inside the pub binary).

As for making this function public, I couldn't think of a reason we would ever need to call this function (other than in tests) in another library, and if we later do, we can always mark remove the annotation. The main drawback to using `@visibleForTesting` is that it is effectively public to any downstream users who don't gate their code on the analyzer, but we don't have to worry about that with the tool.